src/main/java/dev/andrylat/vomelianchuk/CardNumberOperations/CardNumberProcess.java

```diff
@@ -0,0 +1,118 @@
+package dev.andrylat.vomelianchuk.CardNumberOperations;
+
+import dev.andrylat.vomelianchuk.CardNumberOperations.enums.Errors;
+import dev.andrylat.vomelianchuk.CardNumberOperations.enums.PaymentSystem;
+
+import java.util.Optional;
+
+public class CardNumberProcess {
+    private static final int LENGTH_CARD_NUMBER = 16;
+    private static final String HIATUS_SEPARATOR = "\\s";
+    private static final int TEN = 10;
+
+    private String cardNumber;
+    private PaymentSystem paymentSystem;
+    private Errors error;
+
+    public CardNumberProcess(String cardNumber) {
+        this.cardNumber = cardNumber;
+    }
+
+    public String operate() {
```

Comment:
It is the worth way you could do.

Your "String" contains ANYTHING! and it is not logical that the method will return error. name doesn't say about it. I can think that as a result I can receive name of payment system, or error, or something else.

Reply:
and as I see it is done as I said. Why you return payment system as string? The code is not reusable in this case. and nobody knows when there is error in the returning value and when there is actual result. 

Don't think about code as for one place using. Always try to make maximum reusable code if it consumes not much time and effort.

Reply:
Fixed here: 1783555